src/lib/ui/Menu/Menu.tsx

```diff
@@ -0,0 +1,179 @@
+import classNames from 'classnames'
+import {
+  createContext,
+  useCallback,
+  useContext,
+  useEffect,
+  useLayoutEffect,
+  useState,
+} from 'react'
+import { createPortal } from 'react-dom'
+
+export interface MenuProps {
+  children: JSX.Element | JSX.Element[]
+  verticalOffset?: number
+  horizontalOffset?: number
+  edgeOffset?: number
+  button: (props: {
```

Comment:
@mikewuu Should we call this `renderButton`? It confused for a while until I read all the code.